code/modules/mob/mob.dm

```diff
@@ -1232,7 +1232,15 @@
 **/
 /mob/proc/has_light_nearby(light_amount = LIGHTING_TILE_IS_DARK)
 	var/turf/mob_location = get_turf(src)
-	return mob_location.get_lumcount() > light_amount
+	var/area/mob_area = get_area(src)
+
+	if(mob_location.get_lumcount() > light_amount)
+		return TRUE
+	else if(!mob_area.static_lighting)
+		return TRUE
```

Comment:
```suggestion
	if(!mob_area.static_lighting || mob_location.get_lumcount() > light_amount)
		return TRUE
```

Why not do it like this?

Reply:
Yes it means the opposite of what you think

Reply:
I'm aware of ternary operators, but you should be careful with nesting them too deeply or if it's involving complex code logic.

Reply:
that isn't a ternary operator